### PR TITLE
Add escaping for translate strings

### PR DESCRIPTION
**Pull request by burnjohn**

### CODE REVIEW

**Comment by stefanor** [comment]
Why was this necessary in templategallery, and not elsewhere?

Will it not cause issues for other applications that we're using this for?

**Comment by burnjohn** [comment]
We have the issue with unescaped strings after babel code transpilling of react. The only change that is there is escaping. The only case when it can influence other apps is handlebars  `href` tag. 

Now we're checking it.

Maybe it's better to bumb major version?

Here is what that function does `Converts the characters "&", "<", ">", '"', and "'" in string to their corresponding HTML entities.`

Link to the docs: https://lodash.com/docs/4.17.10#escape

**Comment by burnjohn** [comment]
So we tried to test these changes but it is quite hard for us to check every translation.  

As I see escaping can break inserting tags to html. And it is better to avoid using this fix with old repositories. 

So taking this risk into account I can propose several options:  


1. Create a major version with escaping translations text
1. Use a dev version of this library for template gallery a/b testing and delete it after new react translation flow will be set up
1. Escape 2 translation files in template gallery by hands and fix it later with new react translation flow 

@stefanor What is the best option for you?

File: lib/translators/javascript.js [hunk 1]
```diff
@@ -10,7 +12,8 @@ function translateJavascript(copy, gettext) {
     needle = match[0];
     msgid = match[1] || match[2];
     bookend = match[1] ? '"' : '\'';
-    replacement = bookend + gettext(msgid) + bookend;
+    var escapedString = lodash.escape(gettext(msgid));
+    replacement = bookend + escapedString + bookend;
```
Comment (stefanor) [inline lib/translators/javascript.js:16]: I don't think it makes sense to have this library substitute an unescaped msgid with an escaped version. That's way too confusing for the user.

The problem that this PR is trying to address is strings that contain quotation marks. I think a simpler solution to those would be to embed the `unescape()` call in the outputted JS.

I could imagine doing something like:
```js
replacement = gettext(msgid);
if (replacement.indexOf(bookend) !== -1) {
  replacement = "unescape('" + escape(replacement) + "')";
} else {
  replacement = bookend + escapedString + bookend;
}
```

Obviously if we were to do something like that, it'd need a bunch of tests, too :)

**Comment by burnjohn** [comment]
Close this due to quotes fix in https://github.com/yola/templategallery/pull/437

**Pull request closed without merging**